src/highlight/highlight.ts

```diff
@@ -1,59 +1,35 @@
-import { Component, Input } from '@angular/core';
-import { bem, getPropertyByPath } from '../utils';
+import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
+
+import { Hit } from 'instantsearch.js';
+
+import { highlight } from 'instantsearch.js/es/helpers';
+import { getPropertyByPath } from 'instantsearch.js/es/lib/utils';
 
 @Component({
   selector: 'ais-highlight',
-  template: `<span [class]="cx()" [innerHtml]="content"></span>`,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `<span class="ais-Highlight" [innerHtml]="content"></span>`,
 })
 export class NgAisHighlight {
   @Input() attribute: string;
-  @Input() hit: { _highlightResult?: {}; label?: string; highlighted?: string };
-  @Input() tagName: string = 'em';
-
-  cx = bem('Highlight');
+  @Input() hit: Partial<Hit>;
+  @Input() tagName: string = 'mark';
 
   get content() {
-    if (this.attribute === 'highlighted') {
-      return this.hit.highlighted
-        ? this.replaceWithTagName(this.hit.highlighted)
-        : this.hit.label;
-    }
-
-    if (this.hit.hasOwnProperty('_highlightResult')) {
-      const attributeHighlighted = getPropertyByPath(
-        this.hit._highlightResult,
-        this.attribute
-      );
-
-      // check that the attributeHighlighted is a string
-      if (
-        attributeHighlighted !== undefined &&
-        typeof attributeHighlighted.value === 'string'
-      ) {
-        return this.replaceWithTagName(attributeHighlighted.value);
-      }
-    }
-
+    const highlightAttributeResult = getPropertyByPath(
+      this.hit._highlightResult,
+      this.attribute
+    );
     const fallback = getPropertyByPath(this.hit, this.attribute);
-    if (!fallback) {
-      console.warn(
-        `Could not find attribute [${
-          this.attribute
-        }] into hit object, will display an empty string.`
-      );
 
-      return '';
+    if (!highlightAttributeResult && fallback) {
```

Comment:
before here, maybe we add an `@MAJOR` comment to say that when InstantSearch falls back to the attribute, we no longer need to manually do that in Angular InstantSearch?